nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1375,6 +1388,18 @@ def forward(
             if self.transformer_block_type in ['pre_ln', 'normformer']:
                 hidden_states = self.input_layernorm(hidden_states)
 
+            if self.is_adapter_available():
```

Comment:
Instead of these extra hooks, why not just make the attention class itself support adapters - specifically IA adapters and then it just becomes a simple add adapter call with no need to modify signature of functions

Reply:
Discussed an approach with @titu1994 offline which I'll be implementing in a separate PR.